src/wrapped_radix_engine_toolkit/high_level/sargon_specific_types/assets_transfers/per_asset/per_asset_fungible_resource.rs

```diff
@@ -0,0 +1,85 @@
+use crate::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, uniffi::Record)]
+pub struct PerAssetFungibleResource {
+    pub resource_address: ResourceAddress,
+    pub divisibility: Option<i32>,
```

Comment:
used for rounding, potentially we can require wallets to already have rounded, but I think this is safer.